packages/react-ui-core/src/DatePicker/__tests__/DatePicker-test.js

```diff
@@ -1,86 +1,278 @@
 import React from 'react'
 import { shallow } from 'enzyme'
-import addDays from 'date-fns/add_days'
+import format from 'date-fns/format'
 import DatePicker from '../DatePicker'
+import Calendar from '../Calendar'
+import Input from '../../Form/Input'
+
+const DATE_FORMAT = 'MM-dd-YYYY'
 
 describe('DatePicker', () => {
-  it('calendar should be hidden by default', () => {
-    const datepicker = shallow(<DatePicker />)
-    expect(datepicker.state().showCalendar).toBeFalsy()
+  const setup = props => (
+    shallow(
+      <DatePicker
+        {...props}
+      />
+    )
+  )
+
+  describe('value', () => {
+    it('there is no date set when no `value` passed', () => {
+      const wrapper = setup()
+      expect(wrapper.state('value')).toBeUndefined()
+    })
+
+    it('uses `value` and formats it correctly when passed in and valid', () => {
+      const value = '10-10-2025'
+      const wrapper = setup({ value })
+      expect(wrapper.state('value')).toEqual('10/10/2025')
+    })
+  })
+
+  describe('readOnly', () => {
+    it('passes `false` by default to Input', () => {
+      const wrapper = setup()
+      expect(wrapper.find(Input).prop('readOnly')).toBe(false)
+    })
+
+    it('passes `true` into Input when set', () => {
+      const wrapper = setup({ readOnly: true })
+      expect(wrapper.find(Input).prop('readOnly')).toBe(true)
+    })
   })
 
-  it('input should be readOnly by default', () => {
-    const datepicker = shallow(<DatePicker />)
-    const input = datepicker.find('input')
+  describe('initial state', () => {
+    it('hides the calendar by default', () => {
+      const wrapper = setup()
+      expect(wrapper.state('showCalendar')).toBe(false)
+    })
+
+    describe('when `value` is a valid date', () => {
+      const values = ['11/11/2025', '9/9/2025', '9-9-2025', '11-11-2025']
+
+      values.forEach(value => {
+        it(`formats ${value} to ${DATE_FORMAT} and sets 'isValidDate' to true`, () => {
+          const wrapper = setup({ value, dateFormat: DATE_FORMAT })
+          const date = format(new Date(value), DATE_FORMAT)
+
+          expect(wrapper.state('value')).toEqual(date)
+          expect(wrapper.state('isValidDate')).toBe(true)
+        })
+      })
+    })
 
-    expect(input.props().readOnly).toBeTruthy()
+    it('does not format the `value` and sets `isValidDate` to false when value is not valid', () => {
+      const value = '111-111-11'
```

Comment:
What do you think about using more of an "edge casey" value, like '1/32/2018'?

Reply:
That's fine, I can do that 👍